includes/class-wc-webhook.php

```diff
@@ -103,6 +111,9 @@ public function process( $arg ) {
 			return;
 		}
 
+		// Mark this $arg as processed to ensure it doesn't get processed again within the current request.
+		$this->processed[] = $arg;
```

Comment:
Why not add the `$arg` as a key and then test via `isset`/`array_key_exists`? I guess that should be quite a bit faster, potentially. Maybe unnecessary optimization...

Reply:
To answer my own question, array keys can't be e.g. arrays or objects and $arg is of mixed type, so this way it's safer, especially for hooks tied to actions which can have objects and arrays as arguments. 
The number of hooks running within one request probably won't be large anyway.